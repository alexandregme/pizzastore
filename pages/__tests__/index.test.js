import React from 'react';
import {shallow} from 'enzyme';
import expect from 'expect';
import Index from '../index';
import Layout from '../../components/layout';
import Pizzas from "../../components/pizzas";


describe("Index page", ()=>{
    let mountedIndex;

    beforeEach(()=>{
        mountedIndex = shallow(<Index />);
    });

    it('renders without crashing', () => {
        shallow(<Index />);
        expect(mountedIndex).toMatchSnapshot();
    });

    it('expect to have Layout component', () => {
        expect(mountedIndex.find(Layout)).toHaveLength(1);
    });

    it('expect to have Pizzas component', () => {
      expect(mountedIndex.find(Pizzas)).toHaveLength(1);
    });
});
