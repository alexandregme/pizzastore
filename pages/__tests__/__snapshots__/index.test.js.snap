// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Index page renders without crashing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <IndexPage />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": <React.Fragment>
        <Pizzas
          pizzas={
            Array [
              Object {
                "basePrice": 9.89,
                "maxToppings": 3,
                "name": "small",
                "toppings": Array [
                  "topping1",
                  "topping2",
                  "topping3",
                  "topping4",
                  "topping5",
                  "topping6",
                  "topping7",
                ],
              },
              Object {
                "basePrice": 10.89,
                "maxToppings": 5,
                "name": "medium",
                "toppings": Array [
                  "topping1",
                  "topping2",
                  "topping3",
                  "topping4",
                  "topping5",
                  "topping6",
                  "topping7",
                ],
              },
              Object {
                "basePrice": 13.49,
                "maxToppings": null,
                "name": "large",
                "toppings": Array [
                  "topping1",
                  "topping2",
                  "topping3",
                  "topping4",
                  "topping5",
                  "topping6",
                  "topping7",
                ],
              },
            ]
          }
        />
        <Connect(Cart) />
      </React.Fragment>,
      "title": "Home Page",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <Pizzas
            pizzas={
              Array [
                Object {
                  "basePrice": 9.89,
                  "maxToppings": 3,
                  "name": "small",
                  "toppings": Array [
                    "topping1",
                    "topping2",
                    "topping3",
                    "topping4",
                    "topping5",
                    "topping6",
                    "topping7",
                  ],
                },
                Object {
                  "basePrice": 10.89,
                  "maxToppings": 5,
                  "name": "medium",
                  "toppings": Array [
                    "topping1",
                    "topping2",
                    "topping3",
                    "topping4",
                    "topping5",
                    "topping6",
                    "topping7",
                  ],
                },
                Object {
                  "basePrice": 13.49,
                  "maxToppings": null,
                  "name": "large",
                  "toppings": Array [
                    "topping1",
                    "topping2",
                    "topping3",
                    "topping4",
                    "topping5",
                    "topping6",
                    "topping7",
                  ],
                },
              ]
            }
          />,
          <Connect(Cart) />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "pizzas": Array [
              Object {
                "basePrice": 9.89,
                "maxToppings": 3,
                "name": "small",
                "toppings": Array [
                  "topping1",
                  "topping2",
                  "topping3",
                  "topping4",
                  "topping5",
                  "topping6",
                  "topping7",
                ],
              },
              Object {
                "basePrice": 10.89,
                "maxToppings": 5,
                "name": "medium",
                "toppings": Array [
                  "topping1",
                  "topping2",
                  "topping3",
                  "topping4",
                  "topping5",
                  "topping6",
                  "topping7",
                ],
              },
              Object {
                "basePrice": 13.49,
                "maxToppings": null,
                "name": "large",
                "toppings": Array [
                  "topping1",
                  "topping2",
                  "topping3",
                  "topping4",
                  "topping5",
                  "topping6",
                  "topping7",
                ],
              },
            ],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Symbol(react.fragment),
    },
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": <React.Fragment>
          <Pizzas
            pizzas={
              Array [
                Object {
                  "basePrice": 9.89,
                  "maxToppings": 3,
                  "name": "small",
                  "toppings": Array [
                    "topping1",
                    "topping2",
                    "topping3",
                    "topping4",
                    "topping5",
                    "topping6",
                    "topping7",
                  ],
                },
                Object {
                  "basePrice": 10.89,
                  "maxToppings": 5,
                  "name": "medium",
                  "toppings": Array [
                    "topping1",
                    "topping2",
                    "topping3",
                    "topping4",
                    "topping5",
                    "topping6",
                    "topping7",
                  ],
                },
                Object {
                  "basePrice": 13.49,
                  "maxToppings": null,
                  "name": "large",
                  "toppings": Array [
                    "topping1",
                    "topping2",
                    "topping3",
                    "topping4",
                    "topping5",
                    "topping6",
                    "topping7",
                  ],
                },
              ]
            }
          />
          <Connect(Cart) />
        </React.Fragment>,
        "title": "Home Page",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <Pizzas
              pizzas={
                Array [
                  Object {
                    "basePrice": 9.89,
                    "maxToppings": 3,
                    "name": "small",
                    "toppings": Array [
                      "topping1",
                      "topping2",
                      "topping3",
                      "topping4",
                      "topping5",
                      "topping6",
                      "topping7",
                    ],
                  },
                  Object {
                    "basePrice": 10.89,
                    "maxToppings": 5,
                    "name": "medium",
                    "toppings": Array [
                      "topping1",
                      "topping2",
                      "topping3",
                      "topping4",
                      "topping5",
                      "topping6",
                      "topping7",
                    ],
                  },
                  Object {
                    "basePrice": 13.49,
                    "maxToppings": null,
                    "name": "large",
                    "toppings": Array [
                      "topping1",
                      "topping2",
                      "topping3",
                      "topping4",
                      "topping5",
                      "topping6",
                      "topping7",
                    ],
                  },
                ]
              }
            />,
            <Connect(Cart) />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "pizzas": Array [
                Object {
                  "basePrice": 9.89,
                  "maxToppings": 3,
                  "name": "small",
                  "toppings": Array [
                    "topping1",
                    "topping2",
                    "topping3",
                    "topping4",
                    "topping5",
                    "topping6",
                    "topping7",
                  ],
                },
                Object {
                  "basePrice": 10.89,
                  "maxToppings": 5,
                  "name": "medium",
                  "toppings": Array [
                    "topping1",
                    "topping2",
                    "topping3",
                    "topping4",
                    "topping5",
                    "topping6",
                    "topping7",
                  ],
                },
                Object {
                  "basePrice": 13.49,
                  "maxToppings": null,
                  "name": "large",
                  "toppings": Array [
                    "topping1",
                    "topping2",
                    "topping3",
                    "topping4",
                    "topping5",
                    "topping6",
                    "topping7",
                  ],
                },
              ],
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": Symbol(react.fragment),
      },
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
